src/psyclone/tests/psyGen_test.py

```diff
@@ -3539,35 +3593,56 @@ def test_fparser2astprocessor_parse_array_dimensions_attributes(
     from fparser.two.Fortran2003 import Specification_Part
     from fparser.two.Fortran2003 import Dimension_Attr_Spec
 
+    sym_table = SymbolTable()
     reader = FortranStringReader("dimension(:)")
     fparser2spec = Dimension_Attr_Spec(reader)
-    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec, sym_table)
     assert shape == [None]
 
     reader = FortranStringReader("dimension(:,:,:)")
     fparser2spec = Dimension_Attr_Spec(reader)
-    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec, sym_table)
     assert shape == [None, None, None]
 
     reader = FortranStringReader("dimension(3,5)")
     fparser2spec = Dimension_Attr_Spec(reader)
-    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec, sym_table)
     assert shape == [3, 5]
 
+    sym_table.declare('var1', 'integer', [])
+    reader = FortranStringReader("dimension(var1)")
+    fparser2spec = Dimension_Attr_Spec(reader)
+    shape = Fparser2ASTProcessor._parse_dimensions(fparser2spec, sym_table)
+    assert len(shape) == 1
+    assert shape[0] == sym_table.lookup('var1')
+
+    # Assumed size arrays not supported
     reader = FortranStringReader("dimension(*)")
     fparser2spec = Dimension_Attr_Spec(reader)
     with pytest.raises(NotImplementedError) as error:
-        _ = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+        _ = Fparser2ASTProcessor._parse_dimensions(fparser2spec, sym_table)
     assert "Could not process " in str(error.value)
     assert "Assumed-size arrays are not supported." in str(error.value)
 
-    reader = FortranStringReader("dimension(var1)")
+    # Explicit shape symbols must be integer
+    reader = FortranStringReader("dimension(var2)")
     fparser2spec = Dimension_Attr_Spec(reader)
     with pytest.raises(NotImplementedError) as error:
-        _ = Fparser2ASTProcessor._parse_dimensions(fparser2spec)
+        sym_table.declare("var2", "real", [])
+        _ = Fparser2ASTProcessor._parse_dimensions(fparser2spec, sym_table)
+    assert "Could not process " in str(error.value)
+    assert ("Only scalar integer literals or symbols are supported for "
+            "explicit shape array declarations.") in str(error.value)
+
+    # Explicit shape symbols can only be Literal or Symbol
+    with pytest.raises(NotImplementedError) as error:
+        class UnrecognizedType(object):
+            pass
```

Comment:
I think this will need a docstring - just explainthe reason for the class.

Reply:
Added docstring, does it make sense?